homeassistant/components/media_source/__init__.py

```diff
@@ -0,0 +1,80 @@
+"""The media_source integration."""
+import re
+
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.components.media_player.const import ATTR_MEDIA_CONTENT_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.integration_platform import (
+    async_process_integration_platforms,
+)
+from homeassistant.loader import bind_hass
+
+from .const import DOMAIN, URI_SCHEME, URI_SCHEME_REGEX
+from .models import Media
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the media_source component."""
+    hass.data[DOMAIN] = {}
+    hass.components.websocket_api.async_register_command(websocket_browse_media)
+    await async_process_integration_platforms(
+        hass, DOMAIN, _process_media_source_platform
+    )
+    return True
+
+
+async def _process_media_source_platform(hass, domain, platform):
+    """Process a media source platform."""
+    platform, cb = await platform.async_setup_media_source(hass)
```

Comment:
Feel free to add `cb` to `good-names` in `pylintrc`

Reply:
btw no need to return `platform`, since `domain` is already passed into this method. If we want to future proof this, we might want to pass a function into this method, so that in the future people can register both a browse_media and search_media potentially.

Reply:
Or maybe even better just have `async_setup_media_source` return a `MediaSource` class:

```python
class LocalSource(MediaSource):
    @property
    def name(self) -> str

    async def async_browse_media(self, media_id)
```